src/binaryutils.cpp

```diff
@@ -299,7 +303,7 @@ InferenceEngine::Blob::Ptr createBlobFromMats(const std::vector<cv::Mat>& images
 }
 
 ov::runtime::Tensor createBlobFromMats_2(const std::vector<cv::Mat>& images, const std::shared_ptr<TensorInfo>& tensorInfo, bool isPipeline) {
-    ov::Shape shape = !isPipeline ? tensorInfo->getShape() : getShapeFromImages(images, tensorInfo);
+    ov::Shape shape = !isPipeline ? tensorInfo->getShape_3().getFlatShape() : getShapeFromImages(images, tensorInfo);
```

Comment:
`getFlatShape` will be removed at some point

This is wrong. `GetFlatShape` always takes `Dimension::maximum` value when creating flat vector.
Imagine situation with model of shape `(1,224,224,[1~3])`.
The user sends image with 1 color channel and we validate it correctly.
Then the tensor is created with `GetFlatShape`. The flat shape will be `(1,224,224,3)`, 3x larger than original image `(1,224,224,1)`.

Reply:
getShapeFromImages(images, tensorInfo); always

Reply:
leave // TODO check me